report/sql_tasks/tasks/report/create_from_scratch/01_functions/01_date_functions.sql

```diff
@@ -0,0 +1,56 @@
+-- A function for truncating dates that accepts the same resolutions as
+-- DATE_TRUNC(), but also accepts:
+--
+-- * all_time - Returning an arbitrary date in the past
+-- * academic_year - Truncating to the start of the academic year
+-- * semester - Truncating to the nearest 6 months
+
+CREATE OR REPLACE FUNCTION report.multi_truncate(resolution TEXT, date DATE)
+RETURNS DATE
+IMMUTABLE
+AS $$
+    SELECT CASE
+        WHEN resolution = 'all_time' THEN '1901-01-01'::DATE
+        WHEN resolution = 'academic_year' THEN (DATE_TRUNC('year', date - INTERVAL '6 month') + INTERVAL '6 month')::DATE
+        WHEN resolution = 'semester' THEN (DATE_TRUNC('year', date) + (
+            CASE
+                WHEN EXTRACT('quarter' FROM date) < 3 THEN INTERVAL '0 months'
+                ELSE INTERVAL '6 months'
+            END
+        ))::DATE
+        ELSE DATE_TRUNC(resolution, date)::date
+    END
+$$
+LANGUAGE SQL;
+
+-- A function for representing dates at various resolutions. This will create
+-- a human readable version of a date as follows:
+--
+-- * all_time - The string "All time"
+-- * academic_year - YYYY
+-- * year - YYYY
+-- * semester - YYYY-S
+-- * month - YYYY-MM-DD
+-- * week - YYYY-MM-DD
+
+CREATE OR REPLACE FUNCTION report.present_date(resolution TEXT, date DATE)
```

Comment:
Maybe the full `represent`?, present might be a confusing word to mix with dates.